packages/flutter/test/cupertino/switch_test.dart

```diff
@@ -414,5 +414,27 @@ void main() {
 
     expect(value, isFalse);
   });
+  testWidgets('Switch is opaque when disabled', (WidgetTester tester) async {
+    final Key switchKey = UniqueKey();
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: StatefulBuilder(
+          builder: (BuildContext context, StateSetter setState) {
+            return Center(
+              child: CupertinoSwitch(
+                key: switchKey,
+                value: false,
+                dragStartBehavior: DragStartBehavior.down,
+                onChanged: null,
+              ),
+            );
+          },
+        ),
+      ),
+    );
+
+    expect(tester.widget<Opacity>(find.byType(Opacity).first).opacity, 0.4);
```

Comment:
use .single instead of .first. This way, if in the future, another maintainer accidentally adds something into the widget tree that introduces another Opacity, this test would fail instead of succeeding but actually testing something else.

Reply:
Really appreciate your feedback on this PR. 

I can't figure this one out, I thought some like this would work:
`
expect(tester.widget<Opacity>(find.byType(Opacity).evaluate().single).opacity, 1.0);
`
However, the function is expecting something of type 'finder' not of type 'element'.

Reply:
Oops my bad, Finder didn't actually implement single. It'd have to be 2 lines then:

```dart
expect(find.byType(Opacity), findsOneWidget);
expect(tester.widget<Opacity>(find.byType(Opacity).first).opacity, 1);
```